### ISSUE TITLE
Avoid calling `collect` with unknown pool address

### ISSUE THREAD

**Issue by mhchia** [CreamFi/balancer-core#15]
## TL;DR
Don't call `BFactory.collect(BPool pool)` because
1. This way we avoid any security concerns.
2. We don't need to call it anyway since the exit fee is zero.

## What's wrong
`BPool.drainTotalReserves` should be secure in most of the condition thanks to the guard at https://github.com/CreamFi/balancer-core/blob/9f194f34252068d81d0a233a38e3b1acf03684a9/contracts/BPool.sol#L819
 However, a possible attack is to to make `BFactory` call another malicious contract and then this contract calls `BPool.drainTotalReserves` through a [delegate call](https://solidity.readthedocs.io/en/v0.4.21/introduction-to-smart-contracts.html#delegatecall-callcode-and-libraries). This way, `msg.sender` in `BPool.drainTotalReserves` will be `_factory` and `require` is bypassed. Thus, we should prevent `BFactory` from calling any contract we don't know. Currently, `BFactory` calls other contracts in the following snippets:

1. In `BFactory.newBPool`
https://github.com/CreamFi/balancer-core/blob/f571d8fe4970e156af15953d896959cff641a666/contracts/BFactory.sol#L59

This is fine because `bpool` is created by `BFactory` and the code is known to us.

2. In `BFactory.collect`
https://github.com/CreamFi/balancer-core/blob/f571d8fe4970e156af15953d896959cff641a666/contracts/BFactory.sol#L123-L124

This is possibly manipulated if `pool` is a contract with malicious/malformed `balanceOf` or `transfer`. If it calls any pool's `drainTotalReserves` with `delegatecall`, `totalReserves` can be withdrawn to attackers address.

3. In `BFactory.collectTokenReserves`
https://github.com/CreamFi/balancer-core/blob/f571d8fe4970e156af15953d896959cff641a666/contracts/BFactory.sol#L133

This is fine because we have `require(_isBPool[address(pool)])` guarding us, which is only passed through when  `pool` is one of **our** pools.
